### PR TITLE
SDP protocol: implement parser and logger v4

### PR DESCRIPTION
**Pull request by glongo**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6627

Describe changes:
- Rebase

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1741

### CODE REVIEW

File: rust/src/sdp/logger.rs [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        js.set_string("phone_number", phone_number)?;
+    }
+    if let Some(conn_data) = &msg.connection_data {
+        log_connection_data(conn_data, js)?;
+    }
+    if let Some(bws) = &msg.bandwidths {
+        log_bandwidth(bws, js)?;
+    }
+    js.set_string("time", &msg.time)?;
+    if let Some(repeat_time) = &msg.time_zone {
```
Comment (jlucovsky) [inline rust/src/sdp/logger.rs:59]: s/msg.time_zone/msg.repeat_time/

Comment (victorjulien) [reply rust/src/sdp/logger.rs:59]: also, an SV test? :)

Comment (glongo) [reply rust/src/sdp/logger.rs:59]: you mean an SV test for this specific field?

Comment (victorjulien) [reply rust/src/sdp/logger.rs:59]: yeah that would have caught this

Comment (jufajardini) [reply rust/src/sdp/logger.rs:59]: Would it be possible to have one that includes all (or most) of the fields we see in the logger?

**Review by catenacyber — CHANGES_REQUESTED** [review]
Some changes were requested and a rebase is needed

**Pull request closed without merging**